src/v/transform/logging/log_manager.cc

```diff
@@ -200,7 +214,11 @@ manager<ClockType>::~manager() = default;
 
 template<typename ClockType>
 ss::future<> manager<ClockType>::start() {
-    return _flusher->template start<>(&_log_buffers);
+    std::error_code ec = co_await _client->create_transform_logs_topic();
```

Comment:
I would prefer to wait and create the topic until the first log message is written if that's not too annoying. This helps in the case of standing up a new cluster and this can fail as nodes are getting setup.

Reply:
Not too annoying. I think I still have a WIP branch that does it this way.

Somebody had suggested on the PRD that creating the topic inline with produce could cause undesirable effects upstream (e.g. stalling `flush` while retrying the rpc or w/e).

>can fail as nodes are getting setup

^^ This is sort of more convincing to me tbh

Reply:
Covered by [this thread](https://github.com/redpanda-data/redpanda/pull/16485#discussion_r1481907866)